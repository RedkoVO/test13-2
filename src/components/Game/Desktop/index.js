import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import MenuNavigation from '../../MenuNavigation/Desktop/AsyncMenuNavigation'
import RollTop from '../components/RollTop'
import RollBottom from '../components/RollBottom'
import RollMobile from '../components/RollMobile'
import GameCounter from '../components/GameCounter'
import GameComments from '../components/GameComments'
import Footer from '../../Footer/Desktop/AsyncFooterDesktop'

import styles from './styles'

const Game = ({
  classes,
  handlePlayNow,
  isShowIframe,
  url,
  slider,
  comments,
  handleLogout,
  imgMain
}) => (
  <div className="mainWrapperGame">
    <div className={classes.containerGame}>
      <header className={classes.headerGame}>
        <div className={classes.gameHeaderMain}>
          <MenuNavigation type="game" page="dark" handleLogout={handleLogout} />
          <RollTop imgMain={imgMain} />
          <RollBottom slider={slider} handlePlayNow={handlePlayNow} />
          <RollMobile imgMain={imgMain} handlePlayNow={handlePlayNow} />
        </div>
      </header>
      <main className={classes.gameMain}>
        <GameCounter />
        <GameComments comments={comments} />
      </main>
      <Footer />
    </div>

    {isShowIframe && (
      <React.Fragment>
        <div
          className={classes.overlayIframe}
          onClick={() => handlePlayNow()}
        />
        <div className={classes.wrIframe}>
          <iframe
            id="gameFrame"
            className={classes.gameIframe}
            title="game"
            src={url}
          />
        </div>
      </React.Fragment>
    )}
  </div>
)

Game.propTypes = {
  classes: PropTypes.object,
  handlePlayNow: PropTypes.func,
  isShowIframe: PropTypes.bool,
  url: PropTypes.string,
  slider: PropTypes.array,
  comments: PropTypes.array,
  handleLogout: PropTypes.func,
  imgMain: PropTypes.string
}

export default withStyles(styles)(Game)
