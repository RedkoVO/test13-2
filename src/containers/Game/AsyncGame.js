import Async from '../../components/App/components/Async'

export const AsyncGame = Async({
  loader: () => import(/* webpackChunkName: "game_page" */ './index')
})

export default AsyncGame
