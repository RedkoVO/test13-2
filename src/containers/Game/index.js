import compose from 'recompose/compose'
import { withHandlers, withProps, withState, pure, lifecycle } from 'recompose'

import { clearStorageForLogout } from '../../utils/localStorageUtils'
import gC from '../../constants'

import AsyncGameDesktop from '../../components/Game/Desktop/AsyncGameDesktop'

import SliderItemImg1 from '../../assets/images/sim_game_icon_4clover.png'
import GameRollImg from '../../assets/images/slider_slot_4clover_m.jpg'
import JungleImg1 from '../../assets/images/jungle_game1.png'
import JungleImg2 from '../../assets/images/jungle_game2.png'
import JungleImg3 from '../../assets/images/jungle_game3.png'
import JungleImg4 from '../../assets/images/jungle_game4.png'
import JungleImg5 from '../../assets/images/jungle_game5.png'
import UserImg from '../../assets/images/user.jpg'

/* TODO: TMP solution! */
const tmpSliderArr = [
  {
    id: 1,
    slides: [
      { id: 1, img: SliderItemImg1 },
      { id: 2, img: SliderItemImg1 },
      { id: 3, img: SliderItemImg1 },
      { id: 4, img: SliderItemImg1 },
      { id: 5, img: SliderItemImg1 },
      { id: 6, img: SliderItemImg1 },
      { id: 7, img: SliderItemImg1 }
    ]
  },
  {
    id: 2,
    slides: [
      { id: 1, img: JungleImg1 },
      { id: 2, img: JungleImg3 },
      { id: 3, img: JungleImg4 },
      { id: 4, img: JungleImg5 },
      { id: 5, img: JungleImg4 },
      { id: 6, img: JungleImg3 },
      { id: 7, img: JungleImg5 }
    ]
  }
]
const tmpArrComments = [
  {
    id: 1,
    avatar: UserImg,
    title: 'Ben F.',
    time: '22 May 2018 12:03',
    text:
      'Curabitur tincidunt orci quis odio aliquet placerat. Nullam laoreet justo eros, quis semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank. s semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank.',
    like: 5,
    comment: 7
  },
  {
    id: 2,
    avatar: UserImg,
    title: 'Thomas D.',
    time: '22 May 2018 12:03',
    text:
      'Nulla non pellentesque sem, sed aliquet justo. Praesent egestas purus et bibendum tempus.Interdum et malesuada fames ac ante ipsum primis in faucibus.',
    like: 5,
    comment: 7
  },
  {
    id: 3,
    avatar: UserImg,
    title: 'Thomas D. ',
    time: '22 May 2018 12:03',
    text:
      'Nulla non pellentesque sem, sed aliquet justo. Praesent egestas purus et bibendum tempus.',
    like: 5,
    comment: 7
  }
]
const tmpMainImgForGame = [GameRollImg, JungleImg2]

/* TODO: GOVNO_CODE !!! WORKAROUND !!! TMP SOLUTION !!! WARNING !!! */
const address = localStorage.getItem('address')
const url = `${gC.IFRAME_URL}${address}`
const urlJungle = 'https://slots-demo.netlify.com/'
/* *** */

export default compose(
  lifecycle({
    componentWillMount() {
      window.scrollTo(0, 0)
    }
  }),
  withProps(props => {
    return {
      slider:
        props.match.params.bundle === 'jungle'
          ? tmpSliderArr[1].slides
          : tmpSliderArr[0].slides,
      url: props.match.params.bundle === 'jungle' ? urlJungle : url,
      comments: tmpArrComments,
      imgMain:
        props.match.params.bundle === 'jungle'
          ? tmpMainImgForGame[1]
          : tmpMainImgForGame[0]
    }
  }),
  withState('isShowIframe', 'setShowIframe', false),
  withHandlers({
    handleLogout: ({ history }) => () => {
      clearStorageForLogout()
      history.push('/')
    },
    handlePlayNow: ({ isShowIframe, setShowIframe }) => () => {
      setShowIframe(!isShowIframe)
    }
  }),
  pure
)(AsyncGameDesktop)
