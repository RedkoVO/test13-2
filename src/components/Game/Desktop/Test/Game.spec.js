import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Game from '../'

const slider = [
  {
    id: 1,
    slides: [
      { id: 1, img: 'SliderItemImg1' },
      { id: 2, img: 'SliderItemImg1' },
      { id: 3, img: 'SliderItemImg1' },
      { id: 4, img: 'SliderItemImg1' },
      { id: 5, img: 'SliderItemImg1' },
      { id: 6, img: 'SliderItemImg1' },
      { id: 7, img: 'SliderItemImg1' }
    ]
  },
  {
    id: 2,
    slides: [
      { id: 1, img: 'JungleImg1' },
      { id: 2, img: 'JungleImg3' },
      { id: 3, img: 'JungleImg4' },
      { id: 4, img: 'JungleImg5' },
      { id: 5, img: 'JungleImg4' },
      { id: 6, img: 'JungleImg3' },
      { id: 7, img: 'JungleImg5' }
    ]
  }
]
const comments = [
  {
    id: 1,
    avatar: 'UserImg',
    title: 'Ben F.',
    time: '22 May 2018 12:03',
    text: 'Curabitur tincidunt orci quis odio aliquet placerat. Nullam laoreet justo eros, quis semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank. s semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank.',
    like: 5,
    comment: 7
  },
  {
    id: 2,
    avatar: 'UserImg',
    title: 'Thomas D.',
    time: '22 May 2018 12:03',
    text: 'Nulla non pellentesque sem, sed aliquet justo. Praesent egestas purus et bibendum tempus.Interdum et malesuada fames ac ante ipsum primis in faucibus.',
    like: 5,
    comment: 7
  }
]

it('renders correctly Desktop', () => {
  const tree = renderer.create(
    <MemoryRouter keyLength={0}>
      <Game
        handlePlayNow={() => {}}
        isShowIframe={false}
        slider={slider}
        url={'url'}
        imgMain={'image'}
        comments={comments}
        handleLogouts={() => {}}
        isDesktop={true}
      />
    </MemoryRouter>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly Mobile', () => {
  const tree = renderer.create(
    <MemoryRouter keyLength={0}>
      <Game
        handlePlayNow={() => {}}
        isShowIframe={false}
        slider={slider}
        url={'url'}
        imgMain={'image'}
        comments={comments}
        handleLogouts={() => {}}
        isDesktop={false}
      />
    </MemoryRouter>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})