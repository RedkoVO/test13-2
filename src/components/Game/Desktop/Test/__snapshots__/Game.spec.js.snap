// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Desktop 1`] = `
<div
  className="mainWrapperGame"
>
  <div
    className="Game-containerGame-1"
  >
    <header
      className="Game-headerGame-2"
    >
      <div
        className="Game-gameHeaderMain-3"
      />
    </header>
    <main
      className="Game-gameMain-4"
    >
      <div
        className="GameCounter-root-8"
      >
        <div
          className="GameCounter-counterWins-13"
        >
          <p
            className="GameCounter-counterBiggestWin-14"
          >
            <span
              className="GameCounter-counterValue-18"
            >
              4666.78 dcb
            </span>
             
            <br />
             BIGGEST WIN 
            <br />
             FOR ALL TIME
          </p>
        </div>
        <div
          className="GameCounter-counterHours-9"
        >
          <p
            className="GameCounter-counterHoursVolume-15"
          >
            <span
              className="GameCounter-counterValue-18"
            >
              167.12 dcb
            </span>
             
            <br />
             VOLUME OF 
            <br />
            24 HOURS
          </p>
        </div>
        <div
          className="GameCounter-counterUsers-10"
        >
          <p
            className="GameCounter-counterUsersVolume-11"
          >
            <span
              className="GameCounter-counterValue-18"
            >
              785
            </span>
             
            <br />
             USERS FOR 
            <br />
            LAST 24 HOURS
          </p>
        </div>
        <div
          className="GameCounter-counterContracts-12"
        >
          <p
            className="GameCounter-counterContractsVolume-16"
          >
            <span
              className="GameCounter-counterValue-18"
            >
              20243.12 dcb
            </span>
             
            <br />
             ALL VOLUME OF 
            <br />
             SMART CONTRACT
          </p>
          <br />
          <a
            className="GameCounter-counterContractsLink-17"
            href="/"
            onClick={[Function]}
          >
            0x7fdcd2a1e52f10c28cb7732f46393e297ecadda1
          </a>
        </div>
      </div>
      <div
        className="GameComments-root-19"
      >
        <div
          className="GameComments-commentItem-20 GameComments-comment1-21"
        >
          <a
            className="GameComments-commentUser-22"
            href="/"
            onClick={[Function]}
          >
            <img
              alt=""
              className="GameComments-commentUserPhoto-23"
              src="UserImg"
            />
          </a>
          <div
            className="GameComments-commentTitle-24"
          >
            <span
              className="GameComments-commentTitleText-25"
            >
              Ben F.
               says:
            </span>
            <time
              className="GameComments-commentDate-26"
              dateTime="2018-05-22 13:30"
            >
              22 May 2018 12:03
            </time>
          </div>
          <p>
            Curabitur tincidunt orci quis odio aliquet placerat. Nullam laoreet justo eros, quis semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank. s semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank.
          </p>
          <a
            className="GameComments-commentLike-28"
            href="/"
            onClick={[Function]}
          >
             
            <span
              className="GameComments-likeValue-29"
            >
              5
            </span>
             Like
          </a>
          <a
            className="GameComments-subComment-27"
            href="/"
            onClick={[Function]}
          >
            <span
              className="GameComments-commentValue-30"
            >
              7
            </span>
             Comment
          </a>
        </div>
        <div
          className="GameComments-commentItem-20 GameComments-comment2-31"
        >
          <a
            className="GameComments-commentUser-22"
            href="/"
            onClick={[Function]}
          >
            <img
              alt=""
              className="GameComments-commentUserPhoto-23"
              src="UserImg"
            />
          </a>
          <div
            className="GameComments-commentTitle-24"
          >
            <span
              className="GameComments-commentTitleText-25"
            >
              Thomas D.
               says:
            </span>
            <time
              className="GameComments-commentDate-26"
              dateTime="2018-05-22 13:30"
            >
              22 May 2018 12:03
            </time>
          </div>
          <p>
            Nulla non pellentesque sem, sed aliquet justo. Praesent egestas purus et bibendum tempus.Interdum et malesuada fames ac ante ipsum primis in faucibus.
          </p>
          <a
            className="GameComments-commentLike-28"
            href="/"
            onClick={[Function]}
          >
             
            <span
              className="GameComments-likeValue-29"
            >
              5
            </span>
             Like
          </a>
          <a
            className="GameComments-subComment-27"
            href="/"
            onClick={[Function]}
          >
            <span
              className="GameComments-commentValue-30"
            >
              7
            </span>
             Comment
          </a>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`renders correctly Mobile 1`] = `
<div
  className="mainWrapperGame"
>
  <div
    className="Game-containerGame-1"
  >
    <header
      className="Game-headerGame-2"
    >
      <div
        className="Game-gameHeaderMain-3"
      >
        <ul
          className="MenuNavigationGame-root-33 dark"
        >
          <li>
            <div
              className="MenuNavigationGame-logout-42"
              onClick={[Function]}
            >
              Logout
            </div>
          </li>
          <li>
            <a
              className="MenuNavigationGame-headerMenuItem-34"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigationGame-menuIcon-35 MenuNavigationGame-dashboard-36"
              />
              <br />
               Dashboard
            </a>
          </li>
          <li>
            <a
              className="MenuNavigationGame-headerMenuItem-34"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigationGame-menuIcon-35 MenuNavigationGame-walletIcon-37"
              />
              <br />
               Wallet
            </a>
          </li>
          <li>
            <a
              className="MenuNavigationGame-headerMenuItem-34"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigationGame-menuIcon-35 MenuNavigationGame-shop-38"
              />
              <br />
               Shop
            </a>
          </li>
          <li>
            <a
              className="MenuNavigationGame-headerMenuItem-34"
              href="/games"
              onClick={[Function]}
            >
              <i
                className="MenuNavigationGame-menuIcon-35 MenuNavigationGame-games-39"
              />
              <br />
               Games
            </a>
          </li>
          <li>
            <a
              className="MenuNavigationGame-headerMenuItem-34"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigationGame-menuIcon-35 MenuNavigationGame-friends-40"
              />
              <br />
               Friends
            </a>
          </li>
          <li>
            <a
              className="MenuNavigationGame-headerMenuItem-34"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigationGame-menuIcon-35 MenuNavigationGame-settings-41"
              />
              <br />
               Settings
            </a>
          </li>
        </ul>
        <div
          className="RollTopDesktop-root-46"
        >
          <div
            className="RollTopDesktop-gameRollImg-47"
          >
            <img
              alt=""
              src="image"
            />
          </div>
          <div
            className="RollTopDesktop-gameRollInf-48"
          >
            <div
              className="HeaderGame-root-74"
            >
              <div
                className="HeaderGame-gameUserBlock-75"
              >
                <p
                  className="HeaderGame-gameUserHello-76"
                >
                  Hello, Ben A 
                  <br />
                  <a
                    className="HeaderGame-gameUserVerification-77"
                    href="/"
                    onClick={[Function]}
                  >
                    KYC VERIFICATION
                  </a>
                </p>
                <div
                  className="HeaderGame-gameUserPhoto-78"
                >
                  <img
                    alt=""
                    src="user.jpg"
                  />
                </div>
              </div>
            </div>
            <div
              className="RollTopDesktop-gameRollNext-49"
            >
              <p>
                next roll in:
                 
                <span
                  className="RollTopDesktop-gameRollNextCounter-50"
                >
                  00:00:23.12
                </span>
              </p>
            </div>
            <div
              className="RollTopDesktop-gameRollDescr-51"
            >
              <div
                className="RollTopDesktop-gameRollDescrleft-52"
              >
                <div
                  className="RollTopDesktop-gameRollTitleBlock-53"
                >
                  <p
                    className="RollTopDesktop-gameRollTitle-54"
                  >
                    4Clover
                  </p>
                  <span
                    className="RollTopDesktop-gameRollOnline-55"
                  >
                    online: 234
                  </span>
                  <br />
                  <span
                    className="RollTopDesktop-gameRollCompany-56"
                  >
                    by Fall Inc.
                  </span>
                </div>
                <div
                  className="RollTopDesktop-gameRollText-57"
                >
                  <p>
                    Integer tempor semper sapien sit amet bibendum. Etiam ornare teInteger tempor semper sapien sit amet bibendum. Etiam ornare te et bibendum. Etiam ornare te
                  </p>
                </div>
                <div
                  className="RollTopDesktop-gameRollTags-58"
                >
                  <p>
                    #ROLLED #BESTCRYPTO #ETH 
                  </p>
                </div>
              </div>
              <div
                className="RollTopDesktop-gameRollRating-59"
              >
                <p
                  className="RollTopDesktop-ratingValue-60"
                >
                  <span
                    className="RollTopDesktop-ratingValueNum-61"
                  >
                    6.5
                  </span>
                  <span
                    className="RollTopDesktop-ratingValueFull-62"
                  >
                    /7
                  </span>
                </p>
                <div
                  className="RollTopDesktop-ratingStars-63"
                >
                  <div
                    className="RollTopDesktop-fullStars-64"
                  />
                  <div
                    className="RollTopDesktop-halfStars-65"
                  />
                </div>
                <span
                  className="RollTopDesktop-ratingCategory-66"
                >
                  DICE CATEGORY
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="RollBottomDesktop-root-67"
        >
          <div
            className="RollBottomDesktop-gameRollActions-68"
          >
            <div
              className="RollBottomDesktop-playNowBtn-69"
              onClick={[Function]}
            >
              Play Now
            </div>
            <a
              className="RollBottomDesktop-addBtn-70"
              href="/"
              onClick={[Function]}
            >
              Add to favorites
            </a>
          </div>
          <div
            className="RollBottomDesktop-gameRollSlider-71"
            id="scrollSlider"
            tabIndex="5000"
          >
            <a
              className="RollBottomDesktop-sliderItem-72"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                className="RollBottomDesktop-sliderItemPhoto-73"
              />
            </a>
            <a
              className="RollBottomDesktop-sliderItem-72"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                className="RollBottomDesktop-sliderItemPhoto-73"
              />
            </a>
          </div>
        </div>
      </div>
    </header>
    <main
      className="Game-gameMain-4"
    >
      <div
        className="GameCounter-root-8"
      >
        <div
          className="GameCounter-counterWins-13"
        >
          <p
            className="GameCounter-counterBiggestWin-14"
          >
            <span
              className="GameCounter-counterValue-18"
            >
              4666.78 dcb
            </span>
             
            <br />
             BIGGEST WIN 
            <br />
             FOR ALL TIME
          </p>
        </div>
        <div
          className="GameCounter-counterHours-9"
        >
          <p
            className="GameCounter-counterHoursVolume-15"
          >
            <span
              className="GameCounter-counterValue-18"
            >
              167.12 dcb
            </span>
             
            <br />
             VOLUME OF 
            <br />
            24 HOURS
          </p>
        </div>
        <div
          className="GameCounter-counterUsers-10"
        >
          <p
            className="GameCounter-counterUsersVolume-11"
          >
            <span
              className="GameCounter-counterValue-18"
            >
              785
            </span>
             
            <br />
             USERS FOR 
            <br />
            LAST 24 HOURS
          </p>
        </div>
        <div
          className="GameCounter-counterContracts-12"
        >
          <p
            className="GameCounter-counterContractsVolume-16"
          >
            <span
              className="GameCounter-counterValue-18"
            >
              20243.12 dcb
            </span>
             
            <br />
             ALL VOLUME OF 
            <br />
             SMART CONTRACT
          </p>
          <br />
          <a
            className="GameCounter-counterContractsLink-17"
            href="/"
            onClick={[Function]}
          >
            0x7fdcd2a1e52f10c28cb7732f46393e297ecadda1
          </a>
        </div>
      </div>
      <div
        className="GameComments-root-19"
      >
        <div
          className="GameComments-commentItem-20 GameComments-comment1-21"
        >
          <a
            className="GameComments-commentUser-22"
            href="/"
            onClick={[Function]}
          >
            <img
              alt=""
              className="GameComments-commentUserPhoto-23"
              src="UserImg"
            />
          </a>
          <div
            className="GameComments-commentTitle-24"
          >
            <span
              className="GameComments-commentTitleText-25"
            >
              Ben F.
               says:
            </span>
            <time
              className="GameComments-commentDate-26"
              dateTime="2018-05-22 13:30"
            >
              22 May 2018 12:03
            </time>
          </div>
          <p>
            Curabitur tincidunt orci quis odio aliquet placerat. Nullam laoreet justo eros, quis semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank. s semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank.
          </p>
          <a
            className="GameComments-commentLike-28"
            href="/"
            onClick={[Function]}
          >
             
            <span
              className="GameComments-likeValue-29"
            >
              5
            </span>
             Like
          </a>
          <a
            className="GameComments-subComment-27"
            href="/"
            onClick={[Function]}
          >
            <span
              className="GameComments-commentValue-30"
            >
              7
            </span>
             Comment
          </a>
        </div>
        <div
          className="GameComments-commentItem-20 GameComments-comment2-31"
        >
          <a
            className="GameComments-commentUser-22"
            href="/"
            onClick={[Function]}
          >
            <img
              alt=""
              className="GameComments-commentUserPhoto-23"
              src="UserImg"
            />
          </a>
          <div
            className="GameComments-commentTitle-24"
          >
            <span
              className="GameComments-commentTitleText-25"
            >
              Thomas D.
               says:
            </span>
            <time
              className="GameComments-commentDate-26"
              dateTime="2018-05-22 13:30"
            >
              22 May 2018 12:03
            </time>
          </div>
          <p>
            Nulla non pellentesque sem, sed aliquet justo. Praesent egestas purus et bibendum tempus.Interdum et malesuada fames ac ante ipsum primis in faucibus.
          </p>
          <a
            className="GameComments-commentLike-28"
            href="/"
            onClick={[Function]}
          >
             
            <span
              className="GameComments-likeValue-29"
            >
              5
            </span>
             Like
          </a>
          <a
            className="GameComments-subComment-27"
            href="/"
            onClick={[Function]}
          >
            <span
              className="GameComments-commentValue-30"
            >
              7
            </span>
             Comment
          </a>
        </div>
      </div>
    </main>
    <footer
      className="Footer-root-43"
    >
      <ul
        className="Footer-footerNav-44"
      >
        <li>
          <a
            className="Footer-footerNavItem-45"
            href="/"
            onClick={[Function]}
          >
            Dashboard
          </a>
        </li>
        <li>
          <a
            className="Footer-footerNavItem-45"
            href="/"
            onClick={[Function]}
          >
            Wallet
          </a>
        </li>
        <li>
          <a
            className="Footer-footerNavItem-45"
            href="/"
            onClick={[Function]}
          >
            Shop
          </a>
        </li>
        <li>
          <a
            className="Footer-footerNavItem-45"
            href="/games"
            onClick={[Function]}
          >
            Games
          </a>
        </li>
        <li>
          <a
            className="Footer-footerNavItem-45"
            href="/"
            onClick={[Function]}
          >
            Friends
          </a>
        </li>
        <li>
          <a
            className="Footer-footerNavItem-45"
            href="/"
            onClick={[Function]}
          >
            Settings
          </a>
        </li>
      </ul>
      <span>
        Vault. All Rights Reserved 2018
      </span>
    </footer>
  </div>
</div>
`;
